WireAPI/Sources/WireAPI/APIs/UpdateEventsAPI/Event decoding/UpdateEventDecodingProxy.swift

```diff
@@ -18,7 +18,27 @@
 
 import Foundation
 
-extension UpdateEvent: Decodable {
+/// A wrapper that provides custom decoding of update events from
+/// JSON payloads received from the backed.
+///
+/// The need for this wrapper arises from another need to persist
+/// update events in a database: to persist an update event we
+/// persist its encoded data, from which we need to decode it after
+/// fetching it from the database. This coding and decoding needs
+/// to be symmetric and for this we rely on the automatic conformance
+/// of `UpdateEvent`.
+///
+/// However, we still need to decode the update events from the JSON
```

Comment:
This was not meant to be in the PR but am happy to explain it here. Update events come from the backend and need to be (manually) decoded into `UpdateEvent`. We also will store these update events in the event database, and to do so we need to encode them to Data, then decode them when retrieving them. This encoding and decoding uses the default conformance of `Codable`. But you can't have the default conformance if you also explicitly conform to `Decoable` (for the network models), hence a proxy object that provides the manually decoding.

It was discussed though that public models such as `UpdateEvent` shouldn't conform to `Codable` since changing the model would probably break something on the consumer side. Instead we might create mirror objects on the consumer side.

Reply:
Cool, now I've got it. Thank you!